### PR TITLE
Update haproxy best practices using haproxy2.5.0

### PR DESCRIPTION
**Pull request by Chad20N13**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)
1、移除systemd方式启动、停止haproxy；
2、修改haproxy安装方式；

1、原来使用yum安装haproxy的方式，会导致haproxy版本过旧；

### Which TiDB version(s) do your changes apply to? (Required)
1、当前haproxy2.5.0版本使用在tidb 5.3.0版本上；

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by lilin90** [comment]
@superlzs0476 PTAL

**Review by superlzs0476 — APPROVED** [review]
LGTM

File: best-practices/haproxy-best-practices.md [hunk 1]
```diff
@@ -72,29 +72,42 @@ yum -y install epel-release gcc systemd-devel
...
 
-You can easily use HAProxy to configure and set up a load-balanced database environment. This section shows general deployment operations. You can customize the [configuration file](http://cbonte.github.io/haproxy-dconv/1.9/configuration.html) based on your actual scenario.
+You can easily use HAProxy to configure and set up a load-balanced database environment. This section shows general deployment operations. You can customize the [configuration file](http://cbonte.github.io/haproxy-dconv/2.5/configuration.html) based on your actual scenario.
 
 ### Install HAProxy
 
-1. Use yum to install HAProxy：
+1. Download HAProxy 2.5.0：
+   wget https://github.com/haproxy/haproxy/archive/refs/tags/v2.5.0.zip
+2. unzip haproxy
```
Comment (TomShawn) [inline best-practices/haproxy-best-practices.md:81]:
```suggestion
1. Download the package of the HAProxy 2.5.0 source code:

    {{< copyable "shell-regular" >}}

    ```bash
    wget https://github.com/haproxy/haproxy/archive/refs/tags/v2.5.0.zip
    ```

2. Unzip the package to get the source code:
```

File: best-practices/haproxy-best-practices.md [hunk 2]
```diff
@@ -72,29 +72,42 @@ yum -y install epel-release gcc systemd-devel
...
 
     {{< copyable "shell-regular" >}}
 
     ```bash
-    yum -y install haproxy
+    unzip v2.5.0.zip
     ```
 
-2. Check whether the installation is successful：
+3. make & make install
```
Comment (TomShawn) [inline best-practices/haproxy-best-practices.md:89]:
```suggestion
3. Compile the application from the source code:
```

File: best-practices/haproxy-best-practices.md [hunk 3]
```diff
@@ -72,29 +72,42 @@ yum -y install epel-release gcc systemd-devel
...
+     {{< copyable "shell-regular" >}}
+    ```bash
+    cd haproxy-2.5.0
+    make clean
+    make -j 8 TARGET=linux-glibc  USE_THREAD=1
+    make PREFIX=${/app/haproxy} SBINDIR=${/app/haproxy/bin} install
+    ```
```
Comment (TomShawn) [inline best-practices/haproxy-best-practices.md:96]:
```suggestion

    {{< copyable "shell-regular" >}}

    ```bash
    cd haproxy-2.5.0
    make clean
    make -j 8 TARGET=linux-glibc  USE_THREAD=1
    make PREFIX=${/app/haproxy} SBINDIR=${/app/haproxy/bin} install
    ```

```

File: best-practices/haproxy-best-practices.md [hunk 4]
```diff
@@ -72,29 +72,42 @@ yum -y install epel-release gcc systemd-devel
...
+4. reconfig profile
```
Comment (TomShawn) [inline best-practices/haproxy-best-practices.md:97]:
```suggestion
4. Reconfigure the profile:
```

File: best-practices/haproxy-best-practices.md [hunk 5]
```diff
@@ -72,29 +72,42 @@ yum -y install epel-release gcc systemd-devel
...
+     {{< copyable "shell-regular" >}}
+    ```bash
+    echo 'export PATH=/app/haproxy/bin:$PATH' >> /etc/profile
+    ```
```
Comment (TomShawn) [inline best-practices/haproxy-best-practices.md:101]:
```suggestion
    {{< copyable "shell-regular" >}}

    ```bash
    echo 'export PATH=/app/haproxy/bin:$PATH' >> /etc/profile
    ```
```

File: best-practices/haproxy-best-practices.md [hunk 6]
```diff
@@ -183,29 +195,18 @@ listen tidb-cluster                        # Database load balancing.
 
 ### Start HAProxy
 
-There are two methods to start HAProxy.
-
-Method 1: Execute `haproxy`. `/etc/haproxy/haproxy.cfg` is read by default (recommended).
+Execute `haproxy`. `/etc/haproxy/haproxy.cfg` is read by default (recommended).
```
Comment (TomShawn) [inline best-practices/haproxy-best-practices.md:198]:
```suggestion
To start HAProxy, run `haproxy`. `/etc/haproxy/haproxy.cfg` is read by default (recommended).
```

File: best-practices/haproxy-best-practices.md [hunk 7]
```diff
@@ -183,29 +195,18 @@ listen tidb-cluster                        # Database load balancing.
...
-```bash
-systemctl start haproxy.service
-```
 
 ### Stop HAProxy
 
-There are two methods to stop HAProxy.
-
-Method 1: Use `kill -9`.
+Use `kill -9`.
```
Comment (TomShawn) [inline best-practices/haproxy-best-practices.md:209]:
```suggestion
To stop HAProxy, use the `kill -9` command.
```

**Comment by TomShawn** [comment]
@Chad20N13 Because the version description in Line 17 is outdated. Can we update the description "The latest stable version 2.0.0 was released on August 16, 2019, bringing more excellent features" to something like "In this guide, HAProxy 2.5.0 is used. It is recommended to use the latest stable version. See (the released version page of HAProxy) for details"?

**Comment by Chad20N13** [comment]
> 

ok..

File: best-practices/haproxy-best-practices.md [hunk 8]
```diff
@@ -14,7 +14,7 @@ This document describes best practices for configuration and usage of [HAProxy](
 
 HAProxy is free, open-source software written in C language that provides a high availability load balancer and proxy server for TCP and HTTP-based applications. Because of its fast and efficient use of CPU and memory, HAProxy is now widely used by many well-known websites such as GitHub, Bitbucket, Stack Overflow, Reddit, Tumblr, Twitter, Tuenti, and AWS (Amazon Web Services).
 
-HAProxy is written in the year 2000 by Willy Tarreau, the core contributor to the Linux kernel, who is still responsible for the maintenance of the project and provides free software updates in the open-source community. The latest stable version 2.0.0 was released on August 16, 2019, bringing more [excellent features](https://www.haproxy.com/blog/haproxy-2-0-and-beyond/).
+HAProxy is written in the year 2000 by Willy Tarreau, the core contributor to the Linux kernel, who is still responsible for the maintenance of the project and provides free software updates in the open-source community. The latest stable version 2.5.0 was released on November 23, 2021, bringing more [excellent features](https://www.haproxy.com/blog/announcing-haproxy-2-5/).
```
Comment (lilin90) [inline best-practices/haproxy-best-practices.md:17]:
```suggestion
HAProxy is written in the year 2000 by Willy Tarreau, the core contributor to the Linux kernel, who is still responsible for the maintenance of the project and provides free software updates in the open-source community. In this guide, HAProxy [2.5.0](https://www.haproxy.com/blog/announcing-haproxy-2-5/) is used. It is recommended to use the latest stable version. See [the released version of HAProxy](http://www.haproxy.org/) for details.
```

File: best-practices/haproxy-best-practices.md [hunk 9]
```diff
@@ -72,30 +72,53 @@ yum -y install epel-release gcc systemd-devel
...
+    ```bash
+    cd haproxy-2.5.0
+    make clean
+    make -j 8 TARGET=linux-glibc  USE_THREAD=1
+    make PREFIX=${/app/haproxy} SBINDIR=${/app/haproxy/bin} install
     ```
-    /usr/sbin/haproxy
+
+4. Reconfigure the profile:
+    {{< copyable "shell-regular" >}}
```
Comment (lilin90) [inline best-practices/haproxy-best-practices.md:107]:
```suggestion
4. Reconfigure the profile:

    {{< copyable "shell-regular" >}}
```

File: best-practices/haproxy-best-practices.md [hunk 10]
```diff
@@ -72,30 +72,53 @@ yum -y install epel-release gcc systemd-devel
...
+
+    ```bash
+    echo 'export PATH=/app/haproxy/bin:$PATH' >> /etc/profile
+    ```
+
+5. Check whether the installation is successful：
```
Comment (lilin90) [inline best-practices/haproxy-best-practices.md:113]:
```suggestion
5. Check whether the installation is successful:
```

File: best-practices/haproxy-best-practices.md [hunk 11]
```diff
@@ -72,30 +72,53 @@ yum -y install epel-release gcc systemd-devel
...
+
+    {{< copyable "shell-regular" >}}
+
+    ```bash
+    which haproxy
     ```
 
+
```
Comment (lilin90) [inline best-practices/haproxy-best-practices.md:121]: Delete an extra blank line.
```suggestion

```

File: best-practices/haproxy-best-practices.md [hunk 12]
```diff
@@ -146,10 +169,10 @@ global                                     # Global configuration.
    log         127.0.0.1 local2            # Global syslog servers (up to two).
    chroot      /var/lib/haproxy            # Changes the current directory and sets superuser privileges for the startup process to improve security.
    pidfile     /var/run/haproxy.pid        # Writes the PIDs of HAProxy processes into this file.
-   maxconn     4000                        # The maximum number of concurrent connections for a single HAProxy process.
+   maxconn     256                         # The maximum number of concurrent connections per HAProxy thread.
+   nbthread    48                          # The maximum number of threads. （The upper limit is equal to the number of cpus)
```
Comment (lilin90) [inline best-practices/haproxy-best-practices.md:173]:
```suggestion
   nbthread    48                          # The maximum number of threads. (The upper limit is equal to the number of CPUs)
```

File: best-practices/haproxy-best-practices.md [hunk 13]
```diff
@@ -72,28 +72,51 @@ yum -y install epel-release gcc systemd-devel
...
 
+3. Compile the application from the source code:
+
+    {{< copyable "shell-regular" >}}
+
+    ```bash
+    cd haproxy-2.5.0
+    make clean
+    make -j 8 TARGET=linux-glibc  USE_THREAD=1
+    make PREFIX=${/app/haproxy} SBINDIR=${/app/haproxy/bin} install
```
Comment (lilin90) [inline best-practices/haproxy-best-practices.md:103]:
```suggestion
    make PREFIX=${/app/haproxy} SBINDIR=${/app/haproxy/bin} install  # Replace `${/app/haproxy}` and `${/app/haproxy/bin}` with your custom directories.
```

**Review by lilin90 — APPROVED** [review]
LGTM

**Review by TomShawn — APPROVED** [review]
LGTM

File: best-practices/haproxy-best-practices.md [hunk 14]
```diff
@@ -183,29 +206,18 @@ listen tidb-cluster                        # Database load balancing.
...
-There are two methods to start HAProxy.
-
-Method 1: Execute `haproxy`. `/etc/haproxy/haproxy.cfg` is read by default (recommended).
+To start HAProxy, run `haproxy`. `/etc/haproxy/haproxy.cfg` is read by default (recommended).
 
 {{< copyable "shell-regular" >}}
 
 ```bash
 haproxy -f /etc/haproxy/haproxy.cfg
 ```
```
Comment (TomShawn) [inline best-practices/haproxy-best-practices.md:216]:
```suggestion
```

File: best-practices/haproxy-best-practices.md [hunk 15]
```diff
@@ -72,28 +72,51 @@ yum -y install epel-release gcc systemd-devel
...
     ```
 
+3. Compile the application from the source code:
+
+    {{< copyable "shell-regular" >}}
+
+    ```bash
+    cd haproxy-2.5.0
+    make clean
+    make -j 8 TARGET=linux-glibc  USE_THREAD=1
```
Comment (siddontang) [inline best-practices/haproxy-best-practices.md:102]: please remove the redundant space

```
make -j 8 TARGET=linux-glibc USE_THREAD=1
```

Comment (TomShawn) [reply best-practices/haproxy-best-practices.md:102]: Thanks! The redundant space is removed in https://github.com/pingcap/docs/pull/7082.

**Pull request merged**